### PR TITLE
Remove llop symlink by using a map for llop files

### LINKED ISSUE #579: Error when running the First Example
Upon installing the latest `tinygrad` module on windows. I am not able to run this example 

```python
from tinygrad.tensor import Tensor

x = Tensor.eye(3, requires_grad=True)
y = Tensor([[2.0, 0, -2.0]], requires_grad=True)
z = y.matmul(x).sum()
z.backward()

print(x.grad.numpy()) # dz/dx
print(x.grad.numpy()) # dz/dy
```

Python 3.10.9 64 bit

I am getting this error
```
Traceback (most recent call last):
  File "D:\ML_RESEARCH\TinyGrad\Examples\first_example.py", line 1, in <module>
    from tinygrad.tensor import Tensor
  File "d:\ml_research\tinygrad\tinygrad\tinygrad\__init__.py", line 1, in <module>
    from tinygrad import tensor, nn  # noqa: F401
  File "d:\ml_research\tinygrad\tinygrad\tinygrad\tensor.py", line 7, in <module>
    from tinygrad.lazy import Device, LazyBuffer
  File "d:\ml_research\tinygrad\tinygrad\tinygrad\lazy.py", line 31, in <module>
    Device = _Device()
  File "d:\ml_research\tinygrad\tinygrad\tinygrad\lazy.py", line 27, in __init__
    self._buffers[name] = [cls for cname, cls in inspect.getmembers(importlib.import_module('tinygrad.llops.'+op), inspect.isclass) if (cname.upper() == name + "BUFFER")][0]
  File "C:\Users\raghv\AppData\Local\Programs\Python\Python310\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "d:\ml_research\tinygrad\tinygrad\tinygrad\llops\ops_triton.py", line 1
    ../../accel/triton/ops_triton.py
    ^
SyntaxError: invalid syntax
```

### PR DESCRIPTION
**Pull request by martinloretzzz**
Fixes #579

This also makes the way how llop's are imported a lot more clearer, by removing all the string manipulation magic. Adds 3 lines.

### CODE REVIEW

**Comment by martinloretzzz** [comment]
Causes "ImageNet to C Compile Test" to fail, because the file starts like this:
image
Should I refactor the compile_efficientnet.py to write out a file recognize.c file and not use shell redirections? Or fix it in another way?

**Comment by geohot** [comment]
Actually, if they are going to be listed out explicitly, this should probably be refactored to be non dynamic.

File: tinygrad/lazy.py [hunk 1]
```diff
@@ -18,16 +18,20 @@
...
-    for op in [os.path.splitext(x)[0] for x in sorted(os.listdir(os.path.join(os.path.dirname(os.path.realpath(__file__)), "llops"))) if x.startswith("ops_")]:
-      name = op[len("ops_"):].upper()
+    for name, path in backends.items():
       if os.environ.get(name, 0) == "1": self.DEFAULT = name  # note: DEFAULT can be a Device that can't be imported. better than silent use of a different device
       try:
-        self._buffers[name] = [cls for cname, cls in inspect.getmembers(importlib.import_module('tinygrad.llops.'+op), inspect.isclass) if (cname.upper() == name + "BUFFER")][0]
+        self._buffers[name] = [cls for cname, cls in inspect.getmembers(importlib.import_module(path), inspect.isclass) if (cname.upper() == name + "BUFFER")][0]
         self.__setattr__(name, name)
       except ImportError as e:  # NOTE: this can't be put on one line due to mypy issue
-        print(op, "not available", e, file=sys.stderr)
```
[the remark this answers is not in the document]
Comment (martinloretzzz) [reply tinygrad/lazy.py:30]: Totally missed that 👍

**Comment by martinloretzzz** [comment]
Can we do that in a way without having a try/catch block around every import?

image

**Comment by geohot** [comment]
Yea, I think you can use a `try_import` function or something.

**Comment by geohot** [comment]
I got sniped by this:

```
def get_buffer(name, base='tinygrad.llops'):
  try:
    return [cls for cname, cls in inspect.getmembers(importlib.import_module(f'{base}.ops_{name}'), inspect.isclass) if (cname.lower() == name + "buffer")][0]
  except ImportError as e:  # NOTE: this can't be put on one line due to mypy issue
    print(name, base, "not available", e, file=sys.stderr)

class _Device:
  def __init__(self) -> None:
    self._buffers : Dict[str, Type[DeviceBuffer]] = {
      'CPU': get_buffer('cpu'), 'GPU': get_buffer('gpu'),
      'LLVM': get_buffer('llvm'), 'TORCH': get_buffer('torch'),
      'TRITON': get_buffer('triton', 'accel.triton')}
    self.DEFAULT : str = "CPU"
    for name in self._buffers:
      if self._buffers[name] is None: continue
      if getenv(name) == 1: self.DEFAULT = name  # note: DEFAULT can be a Device that can't be imported. better than silent use of a different device
      self.__setattr__(name, name)
Device = _Device()
```

**Comment by geohot** [comment]
Merged, let's see if I broke the tests.

https://github.com/geohot/tinygrad/commit/ac8daaeea540f0de8d0af33b44245d7478fee7ab

**Comment by martinloretzzz** [comment]
Nice✨

**Pull request closed without merging**